qutip/tests/test_floquet.py

```diff
@@ -67,6 +70,94 @@ def testFloquetUnitary(self):
+
+        H0 = - delta / 2.0 * sigmax() - eps0 / 2.0 * sigmaz()
+        H1 = A / 2.0 * sigmax()
+        args = {'w': omega}
+        H = [H0, [H1, lambda t, args: np.sin(args['w'] * t)]]
+
+        # noise power spectrum
+        gamma1 = 0.1
+        def noise_spectrum(omega):
+            return 0.5 * gamma1 * omega / (2 * np.pi)
+
+        # find the floquet modes for the time-dependent hamiltonian
+        f_modes_0, f_energies = floquet_modes(H, T, args)
+
+        # solve the floquet-markov master equation
+        output = fmmesolve(H, psi0, tlist,
+                           [sigmax()], [num(2)], [noise_spectrum], T, args)
+
+        # calculate expectation values in the computational basis
+        p_ex = output.expect[0]
+
+        # For reference: calculate the same thing with mesolve
+        output = mesolve(H, psi0, tlist, [np.sqrt(gamma1) * sigmax()],
+                         [num(2)], args)
+        p_ex_ref = output.expect[0]
+
+        assert_(max(abs(np.real(p_ex) - np.real(p_ex_ref))) < 1e-1)
+
+    def testFloquetDissipative(self):
+        """
+        Floquet: test dissipative evolution of time-dependent two-level system
+        """
+        delta = 0.0 * 2 * np.pi
+        eps0 = 1.0 * 2 * np.pi
+        A = 0.25 * 2 * np.pi
+        omega = 1.0 * 2 * np.pi
+        T = (2 * np.pi) / omega
+        tlist = np.linspace(0.0, 20 * T, 101)
+        psi0 = basis(2, 0)
+
+        H0 = - delta / 2.0 * sigmax() - eps0 / 2.0 * sigmaz()
+        H1 = A / 2.0 * sigmax()
+        args = {'w': omega}
+        H = [H0, [H1, lambda t, args: np.sin(args['w'] * t)]]
+
+        # noise power spectrum
+        gamma1 = 0.1
+        def noise_spectrum(omega):
+            return 0.5 * gamma1 * omega / (2 * np.pi)
+
+        # find the floquet modes for the time-dependent hamiltonian
+        f_modes_0, f_energies = floquet_modes(H, T, args)
+
+        # precalculate mode table
+        f_modes_table_t = floquet_modes_table(f_modes_0, f_energies,
+                                              np.linspace(0, T, 500 + 1),
+                                              H, T, args)
+
+        # solve the floquet-markov master equation
+        output = fmmesolve(H, psi0, tlist,
+                           [sigmax()], [], [noise_spectrum], T, args)
+
+        # calculate expectation values in the computational basis
+        p_ex = np.zeros(np.shape(tlist), dtype=np.complex)
+        for idx, t in enumerate(tlist):
+            f_modes_t = floquet_modes_t_lookup(f_modes_table_t, t, T)
+            p_ex[idx] = expect(num(2), output.states[idx].transform(f_modes_t,
+                                                                    True))
+
+        # For reference: calculate the same thing with mesolve
+        output = mesolve(H, psi0, tlist, [np.sqrt(gamma1) * sigmax()],
+                         [num(2)], args)
+        p_ex_ref = output.expect[0]
+
+        assert_(max(abs(np.real(p_ex) - np.real(p_ex_ref))) < 1e-1)
```

Comment:
Not sure exactly what happens with these computations.

In QuTiP 4.2.0 (installed from `pip`),

```
> max(abs(np.real(p_ex) - np.real(p_ex_ref)))
0.022044446457344991
```

whereas in QuTiP 4.3.0 (from the github `master` branch, installed using `pip install .`),

```
> max(abs(np.real(p_ex) - np.real(p_ex_ref)))
 0.098117562402980751
```

Not sure exactly what changed between these versions (but as far as I could test, this is pre-existing in the `master` branch before this PR).

Reply:
This code is not working for other hamiltonians. Does it support two level systems only?